waku/v2/waku_discv5.nim

```diff
@@ -48,6 +48,36 @@ type WakuDiscoveryV5* = ref object
     protocol*: protocol.Protocol
     listening*: bool
 
+func topicsToRelayShards*(topics: seq[string]): Result[Option[RelayShards], string] =
+  let parsedTopicsRes = topics.mapIt(NsPubsubTopic.parse(it))
+
+  var prevTopic: NsPubsubTopic
+  for i, topic in parsedTopicsRes:
+    if topic.isErr():
+      return err($topic.error)
+
+    if i == 0:
+      prevTopic = topic.value
+      continue
+
+    if prevTopic.kind != topic.value.kind:
+      return err("use named topics OR sharded ones not both.")
+
+    if prevTopic.kind == NsPubsubTopicKind.StaticSharding:
+      if prevTopic.cluster != topic.value.cluster:
+        return err("use sharded topics within the same cluster.")
+
+    if i > 0:
+      prevTopic = topic.value
+
+  if prevTopic.kind == NsPubsubTopicKind.StaticSharding:
+    let indices = parsedTopicsRes.mapIt(it.value.shard)
+    let relayShards = RelayShards.init(prevTopic.cluster, indices)
+
+    return ok(some(relayShards))
```

Comment:
Good idea to have the final `return` be a default and not subject to a condition. In other words, the "happy path" for the function returns the final return statement. Every other return is either an error condition or a quick success.